umbra-js/test/Umbra.test.ts

```diff
@@ -176,172 +206,248 @@ describe('Umbra class', () => {
+          usedReceivers = receivers.slice(0, 5);
+        }
+        if (tx) await tx.wait();
+
+        for (let i = 0; i < usedReceivers.length; i++) {
+          const receiver = usedReceivers[i];
+          const stealthKeyPair = stealthKeyPairs[i];
+          // RECEIVER
+          // Receiver scans for funds sent to them
+          const { userAnnouncements } = await umbra.scan(receiver.publicKey, receiver.privateKey);
+          expect(userAnnouncements.length).to.be.greaterThan(0);
+
+          // Withdraw (test regular withdrawal, so we need to transfer ETH to pay gas)
+          // Destination wallet should have a balance equal to amount sent.
+          const destinationWallet = ethers.Wallet.createRandom();
+          // Fund stealth address to can pay for gas.
+          await sender.sendTransaction({ to: stealthKeyPair.address, value: parseEther('1') });
+
+          // Now we withdraw the tokens
+          const stealthPrivateKey = Umbra.computeStealthPrivateKey(
+            receiver.privateKey,
+            userAnnouncements[0].randomNumber
+          );
+          verifyEqualValues(await dai.balanceOf(destinationWallet.address), 0);
+          const withdrawTxToken = await umbra.withdraw(stealthPrivateKey, dai.address, destinationWallet.address);
+          await withdrawTxToken.wait();
+          verifyEqualValues(await dai.balanceOf(destinationWallet.address), quantity);
+          verifyEqualValues(await dai.balanceOf(stealthKeyPair.address), 0);
+
+          // And for good measure let's withdraw the rest of the ETH
+          const initialEthBalance = await getEthBalance(stealthKeyPair.address);
+          const withdrawTxEth = await umbra.withdraw(stealthPrivateKey, ETH_ADDRESS, destinationWallet.address);
+          await withdrawTxEth.wait();
+          const withdrawEthReceipt = await ethersProvider.getTransactionReceipt(withdrawTxEth.hash);
+          const withdrawTokenTxCost = withdrawEthReceipt.gasUsed.mul(withdrawEthReceipt.effectiveGasPrice);
+          verifyEqualValues(await getEthBalance(stealthKeyPair.address), 0);
+          verifyEqualValues(
+            await getEthBalance(destinationWallet.address),
+            BigNumber.from(initialEthBalance).sub(withdrawTokenTxCost)
+          );
+        }
+      });
+
+      it('Send tokens, scan for them, withdraw them (relayer withdraw)', async () => {
+        // SENDER
+        // Mint Dai to sender, and approve the Umbra contract to spend their DAI
+        await mintAndApproveDai(sender, sender.address, quantity.mul(5));
+
+        let tx: ContractTransaction | null = null;
+        let stealthKeyPairs: KeyPair[] = [];
+        let usedReceivers: Wallet[] = [];
+
+        if (test.id === 'send') {
+          const result = await umbra.send(sender, dai.address, quantity, receiver!.publicKey, overrides);
+          tx = result.tx;
+          stealthKeyPairs = [result.stealthKeyPair];
+          usedReceivers = receivers.slice(0, 1);
```

Comment:
Similar comments to above for this test

Reply:
Answered here: https://github.com/ScopeLift/umbra-protocol/pull/426#discussion_r1158327028

Reply:
Addressed here: https://github.com/ScopeLift/umbra-protocol/pull/426/commits/ddc768e7504ccb601877993210dbfa445f644277